src/axom/sidre/core/Group.cpp

```diff
@@ -25,6 +23,14 @@ namespace axom
 namespace sidre
 {
 
+// Helper macro for defining a prepend string for sidre::Group log messages
+// We are using it to add the pathName() of the group
+#ifndef SIDRE_GROUP_LOG_PREPEND
+# define SIDRE_GROUP_LOG_PREPEND                             \
```

Comment:
Note: I implemented this PR by defining macros ``SIDRE_{GROUP,VIEW}_LOG_PREPEND`` (here and in ``View.hpp``) that I consistently added to the ``SLIC_*_MSG`` calls.

One concern that I have with this approach is that there is nothing to enforce/maintain consistency when new logging/check macros are added to sidre. My hope is that the existing examples (and utility of this functionality) will be sufficient.

I chose this route, rather than defining a new set of macros, e.g. ``SIDRE_*_MSG`` (which would wrap ``SLIC_*_MSG`` and would likely live in a separate file). The latter would introduce new macros that we would need to document and which a developer would have to discover/understand before using.

I think the proposed approach has a nice potential advantage for source-based debugging of Sidre (e.g. from user applications): A developer can easily modify the ``PREPEND`` string to add additional information about the groups/views, like the number of parents/children, or other properties.